pkg/util/metric/metric.go

```diff
@@ -176,10 +185,250 @@ func maybeTick(m periodic) {
 	}
 }
 
+// A HdrHistogram collects observed values by keeping bucketed counts. For
+// convenience, internally two sets of buckets are kept: A cumulative set (i.e.
+// data is never evicted) and a windowed set (which keeps only recently
+// collected samples).
+//
+// Top-level methods generally apply to the cumulative buckets; the windowed
+// variant is exposed through the Windowed method.
+type HdrHistogram struct {
+	Metadata
+	maxVal int64
+	mu     struct {
+		syncutil.Mutex
+		cumulative *hdrhistogram.Histogram
+		*tickHelper
+		sliding *hdrhistogram.WindowedHistogram
+	}
+}
+
+func (h *HdrHistogram) ValueAtQuantileWindowed(q float64) float64 {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	return ValueAtQuantileWindowed(h.ToPrometheusMetricWindowed().Histogram, q)
+}
+
+func (h *HdrHistogram) Mean() float64 {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	return h.mu.cumulative.Mean()
+}
+
+func (h *HdrHistogram) TotalSum() float64 {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	return h.ToPrometheusMetric().GetSummary().GetSampleSum()
+}
+
+// useHdrHistogramsEnvVar can be used to switch all histograms to use the
+// legacy HDR histograms (except for those that explicitly force the use
+// of the newer Prometheus via HistogramOptions.ForceUsePrometheus).
+// HDR Histograms dynamically generate bucket boundaries, which can lead
+// to hundreds of buckets. This can cause performance issues with timeseries
+// databases like Prometheus.
+const useHdrHistogramsEnvVar = "COCKROACH_ENABLE_HDR_HISTOGRAMS"
+
+var hdrEnabled = envutil.EnvOrDefaultBool(useHdrHistogramsEnvVar, false)
```

Comment:
since you're bringing this stuff back, consider making this setting metamorphic:
```
var hdrEnabled = util.ConstantWithMetamorphicTestBool(useHdrHistogramsEnvVar, envutil.EnvOrDefaultBool(....))
```

Reply:
Thanks - I wasn't aware of this function. I changed it over to make use of it.